### PR TITLE
Custom configIO(Issue: MUR-10857)

### PR DESCRIPTION
**Pull request by sauvainr**

### CODE REVIEW

File: modules/configIO.lua [hunk 1]
```diff
@@ -1,45 +1,65 @@
...
+    if result ~= nil and result.value ~= nil and result.value ~= "" then
       -- Following is a VM global value cached on hot VM
-      configIOCache, err = json.parse(result.value)
+      globalConfigIOCache, err = json.parse(result.value)
       if err ~= nil then
-        print("The config_io parse error", err)
+        log.error("'config_io' parsing error", err)
       else
-        configIOCache_ts = now
+        globalConfigIOCache_ts = now -- global
```
Comment (lenaw) [inline modules/configIO.lua]: where is this `global` defined?

Comment (lenaw) [reply modules/configIO.lua]: oh sorry, `--` means comments, i though is an array operator.

File: modules/configIO.lua [hunk 2]
```diff
@@ -1,45 +1,65 @@
...
+  if type(configIO_b) == "string" then configIO_b = json.parse(configIO_b) end
+
+  for k,v in pairs(configIO_a.channels or {}) do config_io[k] = v end
+  for k,v in pairs(configIO_b.channels or {}) do config_io[k] = v end
+  return config_io
 end
 
-function configIO.set(configIO)
+function configIO.set(config_io)
+  if type(config_io) == "string" then json.parse(config_io) end
```
Comment (lenaw) [inline modules/configIO.lua]: `if type(config_io) == "string" then config_io = json.parse(config_io) end`

File: modules/configIO.lua [hunk 3]
```diff
@@ -1,45 +1,65 @@
...
+  config_io.last_edited = os.date("!%Y-%m-%dT%H:%M:%S.000Z", timestamp)
+
+  local configIOTable = { timestamp = timestamp * 1000000, config_io = config_io }
+  if vendorIO and vendorIO.config_io then
+    if vendorIO.timestamp then
+      configIOTable.timestamp = vendorIO.timestamp
+    end
+    configIOTable.config = configIO.merge(vendorIO.config_io, config_io)
+  end
+  if type(configIOTable.config_io) ~= "string" then
```
Comment (lenaw) [inline modules/configIO.lua]: this judgement is redundant, in `line 36` we already made sure `config_io` is not a string.

File: modules/configIO.lua (hunk 1 above)
Comment (sauvainr) [reply modules/configIO.lua]: :) but I added in the name so its not necessary.

File: modules/configIO.lua (hunk 2 above)
Comment (sauvainr) [reply modules/configIO.lua]: thx
```suggestion
  if type(config_io) == "string" then config_io = json.parse(config_io) end
```

**Comment by sauvainr** [comment]
Migration tested..

**Pull request merged**